test/cases/DBVersioner.test.ts

```diff
@@ -0,0 +1,236 @@
+import assert from "node:assert";
+import {
+  DBVersioner,
+  MongoDBAdapter,
+  VERSION_TYPES,
+} from "../../src";
+import { loggerMock } from "../mocks/logger";
+import { dbMock, MongoClientMock } from "../mocks/mongo";
+
+
+describe("DB versioner", function () {
+  this.timeout(5000);
+
+  let mongoAdapter : MongoDBAdapter;
+  let versioner : DBVersioner;
+
+  const contractsVersion = "1.7.9";
+  const dbVersion = "109381236293746234";
+
+  before(async () => {
+
+    versioner = new DBVersioner({
+      dbVersion,
+      contractsVersion,
+      archive: false,
+      logger: loggerMock,
+    });
+
+    mongoAdapter = new MongoDBAdapter({
```

Comment:
not sure why this is needed if you only test versioner.
if you do test connection, you should connect them properly.
Now `versioner` instance is different from the instance created by `mongoAdapter`. so you have 2 different versioner instances in this test.